packages/core/src/useNavigationCache.tsx

```diff
@@ -87,6 +91,13 @@ export default function useNavigationCache<
           ...rest,
           ...helpers,
           ...emitter.create(route.key),
+          dangerouslyGetParent: () =>
+            parentNavigation
```

Comment:
I think this should be `navigation`, not `parentNavigation`. Reading the React navigation docs, this refers to the navigation object of the navigator.